packages/passport/sdk/src/react/PassportContext.tsx

```diff
@@ -0,0 +1,242 @@
+/* eslint-disable @typescript-eslint/naming-convention */
+import React, {
+  createContext, useContext, useEffect, useMemo,
+  useState,
+} from 'react';
+import { Web3Provider } from '@ethersproject/providers';
+import { Passport } from '../Passport';
+import { PassportModuleConfiguration, UserProfile } from '../types';
+
+type PassportContextType = {
+  passportInstance: Passport;
+  isLoggedIn: boolean;
+  login: () => Promise<string[]>;
+  logout: () => Promise<void>;
+  loginWithoutWallet: () => Promise<UserProfile | null>;
+  loginWithEthersjs: () => Promise<{ accounts: string[], provider: Web3Provider | null }>;
+  isLoading: boolean;
+  error: Error | null;
+  accounts: string[];
+};
+
+const PassportContext = createContext<PassportContextType>({
+  passportInstance: {} as Passport,
+  isLoggedIn: false,
+  login: () => { throw new Error('login must be used within a PassportProvider'); },
+  logout: () => { throw new Error('logout must be used within a PassportProvider'); },
+  loginWithoutWallet: () => { throw new Error('loginWithoutWallet must be used within a PassportProvider'); },
+  loginWithEthersjs: () => { throw new Error('loginWithEthersjs must be used within a PassportProvider'); },
+  isLoading: false,
+  error: null,
+  accounts: [],
+});
+
+type PassportProviderProps = {
+  config: PassportModuleConfiguration;
+  children: React.ReactNode;
+};
+
+export function PassportProvider({ children, config }: PassportProviderProps) {
+  const [isLoggedIn, setLoggedIn] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [accounts, setAccounts] = useState<string[]>([]);
+  const [error, setError] = useState<Error | null>(null);
+  const passportInstance = new Passport(config);
+  const v = useMemo(() => ({
+    passportInstance,
+    isLoggedIn,
+    isLoading,
+    accounts,
+    error,
+    login: async () => {
+      try {
+        setError(null);
+        setIsLoading(true);
+        const provider = passportInstance.connectEvm();
+        const acc = await provider.request({ method: 'eth_requestAccounts' });
+        setLoggedIn(true);
+        setAccounts(acc);
+        return accounts;
+      } catch (e: any) {
+        setError(e);
+      } finally {
+        setIsLoading(false);
+      }
+      return [];
+    },
+    logout: async () => {
+      try {
+        setError(null);
+        setIsLoading(true);
+        await passportInstance.logout();
+        setLoggedIn(false);
+        setAccounts([]);
+      } catch (e: any) {
+        setError(e);
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    loginWithoutWallet: async () => {
+      try {
+        setError(null);
+        setIsLoading(true);
+        const profile = await passportInstance.login();
+        setLoggedIn(true);
+        return profile;
+      } catch (e: any) {
+        setError(e);
+        return null;
+      } finally {
+        setIsLoading(false);
+      }
+    },
+    loginWithEthersjs: async () => {
```

Comment:
Passport sdk package already has `@ethersproject/providers` as an dependency tho. Should I change it to peer?

Reply:
We use some of the modules from Ethers internally, but we do not expose Ethers to consumers of Passport. By exposing Ethers from Passport like this, we'll need to continue to support it indefinitely, or make a breaking change if we decide to remove it.

This could also open up the door for partners to request that we add methods for other wallet connection libraries (e.g. Viem), so I think it's best if we just expose the raw Passport provider and let consumers wrap it themselves.

Reply:
I'll try remove this and see what kind of experience it is with 3rd party provider.